packages/manager/modules/anthos/src/dashboard/general-information/general-information.controller.js

```diff
@@ -0,0 +1,77 @@
+import get from 'lodash/get';
+
+export default class {
+  /* @ngInject */
+  constructor($filter, $translate, CucCloudMessage, AnthosTenantsService) {
+    this.$filter = $filter;
+    this.$translate = $translate;
+    this.CucCloudMessage = CucCloudMessage;
+    this.AnthosTenantsService = AnthosTenantsService;
+  }
+
+  computeStoragePercent(value) {
+    return (value / this.netappStorage.totalSize) * 100;
+  }
+
+  getStorageInfo() {
+    const { totalSize, usedSize } = this.netappStorage;
+    const usedStorage = this.$filter('cucBytes')(usedSize, 0, false, 'MB');
+    const totalStorage = this.$filter('cucBytes')(totalSize, 0, false, 'MB');
+    const percentStorage = this.computeStoragePercent(usedSize).toFixed(2);
+    return `${usedStorage} / ${totalStorage} (${percentStorage}%)`;
+  }
+
+  onAnthosRegeneratePasswordClick() {
+    return this.AnthosTenantsService.resetTenantAdminAccess(this.serviceName)
+      .then(({ accessUrl }) => {
+        this.tenant.accessUrl = accessUrl;
+        return this.CucCloudMessage.success(
+          this.$translate.instant(
+            'anthos_tenant_dashboard_general_information_tile_security_center_access_anthos_menu_regenerate_pwd_success',
+          ),
+        );
+      })
+      .catch((error) => {
+        return this.CucCloudMessage.error(
+          this.$translate.instant(
+            'anthos_tenant_dashboard_general_information_tile_security_center_access_anthos_menu_regenerate_pwd_failed',
+            {
+              message: get(error, 'data.message'),
+            },
+          ),
+        );
+      });
+  }
+
+  onNetappRegeneratePasswordClick() {
+    return this.AnthosTenantsService.resetTenantStorageAdminAccess(
+      this.serviceName,
+    )
+      .then(({ accessUrl }) => {
+        this.tenant.storageAccessUrl = accessUrl;
+        return this.CucCloudMessage.success(
+          this.$translate.instant(
+            'anthos_tenant_dashboard_general_information_tile_security_center_access_netapp_menu_regenerate_pwd_success',
+          ),
+        );
+      })
+      .catch((error) => {
+        return this.CucCloudMessage.error(
+          this.$translate.instant(
+            'anthos_tenant_dashboard_general_information_tile_security_center_access_netapp_menu_regenerate_pwd_failed',
+            {
+              message: get(error, 'data.message'),
+            },
+          ),
+        );
+      });
+  }
+
+  onOpenAnthosConsoleClick() {
+    this.openAnthosConsole();
+  }
```

Comment:
Can't we just directly pass the function ?

Reply:
No, because we will have tracking later.

Reply:
Finally removed, regarding others changes.